### PR TITLE
Add missing REST endpoints for karaf distribution

### PR DESCRIPTION
**Pull request by tbouron**
This change add some missing endpoint for the karaf distribution:
- `/v1/catalog/bundles`
- `/v1/catalog/types`

### CODE REVIEW

File: rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml [hunk 1]
```diff
@@ -75,6 +75,7 @@ limitations under the License.
     <bean id="apidocResourceBean" class="org.apache.brooklyn.rest.resources.ApidocResource"/>
     <bean id="applicationResourceBean" class="org.apache.brooklyn.rest.resources.ApplicationResource"/>
     <bean id="catalogResourceBean" class="org.apache.brooklyn.rest.resources.CatalogResource"/>
+    <bean id="bundlesResourceBean" class="org.apache.brooklyn.rest.resources.BundleResource"/>
```
Comment (aledsage) [inline rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml]: also need `typesResourceBean` for `TypeResource`

File: rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml [hunk 2]
```diff
@@ -74,18 +74,20 @@ limitations under the License.
     <bean id="adjunctResourceBean" class="org.apache.brooklyn.rest.resources.AdjunctResource"/>
     <bean id="apidocResourceBean" class="org.apache.brooklyn.rest.resources.ApidocResource"/>
     <bean id="applicationResourceBean" class="org.apache.brooklyn.rest.resources.ApplicationResource"/>
+    <bean id="bundlesResourceBean" class="org.apache.brooklyn.rest.resources.BundleResource"/>
```
Comment (aledsage) [inline rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml]: Very minor: I'd go for `bundleResourceBean` to be consistent with all the others. Then good to merge!

File: rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml (hunk 1 above)
Comment (tbouron) [reply rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml]: Good catch! Endpoint added @aledsage

File: rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml (hunk 2 above)
Comment (tbouron) [reply rest/rest-resources/src/main/resources/OSGI-INF/blueprint/service.xml]: @aledsage updated

**Pull request merged**